src/views/standalone/system/RebootAndShutdownView.vue

```diff
@@ -4,13 +4,166 @@
 -->
 
 <script setup lang="ts">
+import { ubusCall } from '@/lib/standalone/ubus'
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import {
+  NeButton,
+  NeTitle,
+  NeModal,
+  NeInlineNotification,
+  NeProgressBar,
+  getAxiosErrorMessage,
+  NeSkeleton
+} from '@nethserver/vue-tailwind-lib'
+import { onMounted } from 'vue'
+import { ref } from 'vue'
 import { useI18n } from 'vue-i18n'
 
+type requestType = 'poweroff' | 'reboot'
+const REBOOT_WAIT_TIME = 45000
+
 const { t } = useI18n()
+
+const hostname = ref('')
+const loading = ref(true)
+const showShutdownModal = ref(false)
+const showRebootModal = ref(false)
+const isPerformingRequest = ref(false)
+const modalRequestError = ref('')
+const pageError = ref('')
+
+const rebootProgress = ref(0)
+const isRebooting = ref(false)
+
+async function getHostname() {
+  try {
+    let systemInfo = await ubusCall('system', 'board')
+    hostname.value = systemInfo.data.hostname
+    loading.value = false
+  } catch (err: any) {
+    pageError.value = t(getAxiosErrorMessage(err))
+  }
+}
+
+async function performRequest(type: requestType) {
+  try {
+    isPerformingRequest.value = true
+    await ubusCall('ns.power', type)
+
+    if (type == 'reboot') {
+      isRebooting.value = true
+      setRebootTimer()
+    } else {
+      closeModal()
+    }
+  } catch (err: any) {
+    modalRequestError.value = t(getAxiosErrorMessage(err))
+  } finally {
+    isPerformingRequest.value = false
+  }
+}
+
+function closeModal() {
+  showRebootModal.value = false
+  showShutdownModal.value = false
+  modalRequestError.value = ''
+}
+
+function setRebootTimer() {
+  setTimeout(() => {
```

Comment:
In order to avoid memory leaks, let's add a `onUnmounted` callback where we call [clearTimeout](https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout) and [clearInterval](https://developer.mozilla.org/en-US/docs/Web/API/clearInterval)